[Megan Wee] iP

mweeruien opened the pull request:
# Notaro Bot
> “I am Notaro! Bc I am Not-A-Ro bot :>” 
> – Notaro (a reliable source)

Notaro frees your mind of having to remember things you need to do. It's,

* text-based
* easy to learn
* ~~FAST~~ *SUPER* FAST to use

All you need to do is,
1. download it from [here](https://github.com/mweeruien/ip).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE**!

Features:
- [ ] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)


If you Java programmer, you can use it to practice Java too. Here's the main method:
```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

And always remember!
`boolean isNotaroBotForYou = true` 😉

Commit:
Add Gradle support

mweeruien committed:
Added the name (Notaro) and changed the introduction

mweeruien committed:
completed level 1, repeats user input and ends when user inputs "bye"

mweeruien committed:
Level 2 completed. Added todoList variable, pointer variable. Added the command to "list".

mweeruien committed:
Added new class Task

mweeruien committed:
Moved Task to within main, edited code so Level-2 functions would work

mweeruien committed:
Completed level 3, accepts "done" commands from user input

mweeruien committed:
Added classes Deadline, Event, and Todo which extend from Task

mweeruien committed:
Completed Level 4, with previous commands now supporting the Deadline, Todo and Event tasks

mweeruien committed:
Added new exceptions classes DukeExceptions and it's children

mweeruien committed:
Completed Level 5! Implemented the new Exceptions classes to throw an error for the wrong command, or for insufficient details given

mweeruien committed:
Added NotDoneRightException and converted the todoList from a String array to an ArrayList

mweeruien committed:
Completed level 6! Added the delete function

mweeruien committed:
Completed the testing components. Fixed scanner bug.

mweeruien committed:
edited switch statements and list command

mweeruien committed:
added data/duke.txt, and the reading of the file

mweeruien committed:
fixed bugs

mweeruien committed:
Completed Level 7

mweeruien committed:
Completed Level 8

mweeruien committed:
Merge branch 'branch-Level-7'

mweeruien committed:
Merge branch 'branch-Level-8'

# Conflicts:
#	src/main/java/Duke.java

mweeruien committed:
Created and organised packages

mweeruien committed:
Completed Storage, TaskList, and Ui

mweeruien committed:
Completed A-MoreOOP

mweeruien committed:
Created DukeTest.java

mweeruien committed:
Complete A-JUnit, add tests in DukeTest.java

mweeruien committed:
Add some javadocs comments

mweeruien committed:
add javadocs

mweeruien committed:
Complete A-JavaDoc

mweeruien committed:
Complete A-CodingStandard

mweeruien committed:
Complete the find function, complete Level-9

mweeruien committed:
Merge branch 'branch-Level-9'

mweeruien committed:
Merge branch 'branch-A-JavaDoc'

# Conflicts:
#	src/main/java/duke/ui/Ui.java

mweeruien committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/duke/tasks/Deadline.java

mweeruien committed:
Complete more Code Standardisation, Add Command class

sunjc826 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,103 @@
+
+            } else if (cmd.equals("find")) {
+                command.find(user_command, taskList);
+
+
```
Extra large gap here. This can be trimmed.

mweeruien replied:
Hi Jia Cheng, thank you for your reviews! I will improve my code based on your suggestions :)

sunjc826 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,103 @@
+        String cmd; // this is the container for the first word of the command received from the user
+
+        ///// This listens for the commands and interprets them
+        // This part listens for user input and repeats until the command "bye" is identified
```
```suggestion
// Listens for user input and repeats until the command "bye" is identified
```
I think briefer comments would be an improvement to comment quality.

sunjc826 commented on src/main/java/duke/command/Command.java:
```diff
@@ -0,0 +1,171 @@
+     * @throws EmptyDescriptionException
+     * @throws IOException
+     */
+    public void addDeadline(String user_command, String cmd, TaskList taskList)
```
Would it be better if `user_command` is written in lowerUpperCase instead as with all other variables?

sunjc826 commented on src/main/java/duke/storage/Storage.java:
```diff
@@ -0,0 +1,152 @@
+
+            } else {
+                System.out.println("Here's your progress:");
+                Scanner file_sc = new Scanner(FILE);
```
Similar to my previous comment, for consistency sake, `file_sc` can be `fileSc`.

sunjc826 commented on src/main/java/duke/tasks/Deadline.java:
```diff
@@ -0,0 +1,47 @@
+     */
+    @Override
+    public String toString() {
+        return "[D]" + super.toString() + description + " (by: " + dateBy.getDayOfMonth()
+                + " " + dateBy.getMonth().toString() + " " + dateBy.getYear() + ")";
```
Consider using String.format here.

sunjc826 commented on src/main/java/duke/tasks/Task.java:
```diff
@@ -0,0 +1,60 @@
+    /**
+     * Returns the representation of the task to be saved in the text file.
+     *
+     * @return an empty string
```
Not too sure if implementation details (empty string) is necessary here.

sunjc826 reviewed (commented):
I like the documentation a lot. It's very well done and makes code reading pleasant.
Some minor nitpicks including the occasional use of snakecased variables, which can be jarring.
Additionally, String.format may be a cleaner way to formulate a string rather than concatenation.

mweeruien committed:
Change variable names in accordance to coding standards

mweeruien committed:
Merge remote-tracking branch 'origin/add-gradle-support'

mweeruien committed:
Add gradle to build and run Duke

mweeruien committed:
Add checkstyle files

mweeruien committed:
Add checkstyle files

tanqiya commented on src/main/java/duke/tasks/Deadline.java:
```diff
@@ -0,0 +1,47 @@
+import java.time.LocalDate;
+
+public class Deadline extends Task {
+    private final LocalDate dateBy;
```
While I don't think this explicitly violates the coding standard, I think it would be clearer to use a variable name such as `dueDate`.

tanqiya commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,103 @@
+                    break;
+
+                default :
+                    throw new CommandDoesNotExist(userCommand);
```
While I don't think this explicitly violates the coding standard, I think it would be clearer to change the name of this exception to `CommandDoesNotExistException` or perhaps `InvalidCommandException` to make it more apparent that it is an exception.

tanqiya commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,103 @@
+    public void run() throws DukeExceptions, IOException {
+        ui.showWelcome();
+        String userCommand; // this is the container for the full command received from the user
+        String cmd; // this is the container for the first word of the command received from the user
```
While I don't think this explicitly violates the coding standard, I think it would be clearer to have a more descriptive name for this variable to differentiate it from `userCommand`, such as `commandWord`.

tanqiya reviewed (commented):
Code was very neat and good use of JavaDocs throughout! 👍 Could not find any direct violations to the coding standard, but there were a few naming suggestions I left.

mweeruien committed:
Add some basic GUI components

mweeruien committed:
Complete the JavaFX tutorial and implement basic GUI

mweeruien committed:
Add GUI features

mweeruien committed:
Clean up code and add Javadocs for new methods

mweeruien committed:
Merge branch 'branch-Level-10'

mweeruien committed:
Create JAR File and Tidy up Code

mweeruien committed:
Add Assert feature to document important assumptions

No assert features to document important assumptions.

Important assumptions are not being checked and can result in bugs.

Add assert statements to check assumptions. Add assert statements to
ensure the user input passed from function to function is not null.

This prevents failures in the code where user input is necessarily not
null.

mweeruien committed:
Improve Code Quality

Code quality does not adhere to the standards set.

This can cause difficulty in reading the code programmed.

Recheck code for areas that could be abstracted, and abstract
appropriate functions and exceptions.

This improves code abstraction and readability, making it easier to
understand.

mweeruien committed:
Complete extension C-tagging

mweeruien committed:
Merge pull request #2 from mweeruien/branch-A-CodeQuality

Improve Code Quality

mweeruien committed:
Merge branch 'master' of https://github.com/mweeruien/ip into branch-C-Tagging

* 'master' of https://github.com/mweeruien/ip:
  Improve Code Quality

mweeruien committed:
Merge branch 'master' of https://github.com/mweeruien/ip

* 'master' of https://github.com/mweeruien/ip:
  Improve Code Quality

mweeruien committed:
Merge branch 'master' into branch-A-Assertions

mweeruien committed:
Merge branch 'master' into branch-C-Tagging

mweeruien committed:
Merge pull request #1 from mweeruien/branch-A-Assertions

Add Assert feature to document important assumptions

mweeruien committed:
Merge branch 'branch-C-Tagging'

mweeruien committed:
Merge branch 'master' of https://github.com/mweeruien/ip

* 'master' of https://github.com/mweeruien/ip:
  Add Assert feature to document important assumptions

# Conflicts:
#	src/main/java/duke/ui/MainWindow.java

mweeruien committed:
Improve GUI by cropping icons and adding background color

mweeruien committed:
Set theme jekyll-theme-slate

mweeruien committed:
Add User Guide and improve formatting of robot's responses

mweeruien committed:
Merge branch 'master' of https://github.com/mweeruien/ip

mweeruien committed:
Fix formatting issues within user guide

mweeruien committed:
Fix checkstyle issues, Improve README, Refactor storage.load() to improve abstraction and readability.

mweeruien committed:
Fix JUnit Tests

mweeruien committed:
Add Time to Deadline and Event and fix JUnit Testing issues.

mweeruien committed:
Update README.md
